apiserver/facades/controller/migrationtarget/migrationtarget.go

```diff
@@ -263,11 +264,22 @@ func (api *API) CheckMachines(args params.ModelArgs) (params.ErrorResults, error
 		return params.ErrorResults{}, errors.Trace(err)
 	}
 	defer st.Release()
+	backend := credentialcommon.NewPersistentBackend(st.State)
+	// Begin check for manual cloud - we do not want to perform certain checks if the cloud
+	// is manaul
+	var model credentialcommon.Model
+	if model, err = backend.Model(); err != nil {
+		return params.ErrorResults{}, errors.Trace((err))
+	}
+	var cloud cloud.Cloud
+	if cloud, err = backend.Cloud(model.CloudName()); err != nil {
+		return params.ErrorResults{}, errors.Trace((err))
+	}
```

Comment:
you can just get the cloud like this, is more typical of what we do elsewhere
```
	model, err := st.Model();
	if err != nil {
		return params.ErrorResults{}, errors.Trace(err)
	}
	cloud, err := model.Cloud()
	if err != nil {
		return params.ErrorResults{}, errors.Trace(err)
	}
	return credentialcommon.ValidateExistingModelCredential(
		credentialcommon.NewPersistentBackend(st.State),
		context.CallContext(st.State),
		cloud.Type != "manual",
	)
```

Reply:
thanks!